src/Core/Log.re

```diff
@@ -143,17 +174,17 @@ let enableDebugLogging = () =>
 let isDebugLoggingEnabled = () =>
   Logs.Src.level(Logs.default) == Some(Logs.Debug);
 
-let log = (~namespace=?, level, msgf) =>
-  Logs.msg(level, m => msgf(m(~header=?None, ~tags=?namespace)));
-
-let infof = msgf => Logs.info(m => msgf(m(~header=?None, ~tags=?None)));
-let info = msg => infof(m => m("%s", msg));
-
-let debugf = msgf => Logs.debug(m => msgf(m(~header=?None, ~tags=?None)));
-let debug = msgf => debugf(m => m("%s", msgf()));
+let log = (~namespace="Global", level, msgf) =>
+  Logs.msg(level, m => {
+    if (Namespace.isEnabled(namespace)) {
+      let tags = Logs.Tag.(empty |> add(Namespace.tag, namespace));
+      msgf(m(~header=?None, ~tags))
+    }
```

Comment:
This filters on all reporters. We can also set up filtering for just the console reporter, but that will add a bit of overhead because at that point it's wrapped in a tag. Probably a negligible overhead, but still.

Reply:
Sounds reasonable to me!